adapter-jdbc/src/main/java/io/nosqlbench/adapter/jdbc/JDBCSpace.java

```diff
@@ -0,0 +1,206 @@
+        }
+
+        Optional<String> sslCert = cfg.getOptional("sslcert");
+        if (sslCert.isPresent()) {
+            hikariConfig.addDataSourceProperty("sslcert", sslCert.get());
+        } /*else if(sslMode.isPresent() && (!"disable".equalsIgnoreCase(sslMode.get()) || !"allow".equalsIgnoreCase(sslMode.get())) || !"prefer".equalsIgnoreCase(sslMode.get())) {
+            throw new OpConfigError("When sslmode is true, sslcert should be provided.");
+        }*/
+
+        Optional<String> sslRootCert = cfg.getOptional("sslrootcert");
+        if (sslRootCert.isPresent()) {
+            hikariConfig.addDataSourceProperty("sslrootcert", sslRootCert.get());
+        }
+
+        Optional<String> applicationName = cfg.getOptional("applicationName");
+        hikariConfig.addDataSourceProperty("applicationName", applicationName.orElse("NoSQLBench"));
+
+        Optional<Boolean> rewriteBatchedInserts = cfg.getOptional(Boolean.class, "rewriteBatchedInserts");
+        hikariConfig.addDataSourceProperty("rewriteBatchedInserts", rewriteBatchedInserts.orElse(true));
+
+        //Maybe always disable auto-commit since we manage ourselves?
+        Optional<Boolean> autoCommit = cfg.getOptional(Boolean.class, "autoCommit");
+        hikariConfig.setAutoCommit(autoCommit.orElse(false));
+
+        int maximumPoolSize = Integer.parseInt(cfg.getOptional("maximumPoolSize").orElse("40"));
+        hikariConfig.setMaximumPoolSize(maximumPoolSize);
+
+        int keepaliveTime = Integer.parseInt(cfg.getOptional("keepaliveTime").orElse("150000"));
+        hikariConfig.setKeepaliveTime(keepaliveTime);
+
+        HikariDataSource hds = new HikariDataSource(hikariConfig);
+        try {
+            this.connection = hds.getConnection();
+        } catch (Exception ex) {
+            String exp = "Exception occurred while attempting to create a connection using the HikariDataSource";
+            logger.error(exp, ex);
+            throw new RuntimeException(exp, ex);
+        }
+
+        return hds;
+    }
+
+    public static NBConfigModel getConfigModel() {
+        return ConfigModel.of(JDBCSpace.class)
+            .add(Param.defaultTo("url", "jdbc:postgresql:/").setDescription("The connection URL used to connect to the DBMS. Defaults to 'jdbc:postgresql:/'"))
+            .add(Param.defaultTo("serverName", "localhost").setDescription("The host name of the server. Defaults to 'localhost'"))
+            .add(Param.optional("databaseName").setDescription("The database name. The default is to connect to a database with the same name as the user name used to connect to the server."))
+            // See https://github.com/brettwooldridge/HikariCP/tree/dev#gear-configuration-knobs-baby & https://jdbc.postgresql.org/documentation/use/
+            .add(Param.defaultTo("portNumber", 5432).setDescription("The port number the server is listening on. Defaults to the PostgreSQL® standard port number (5432)"))
+            .add(Param.optional("user"))
```

Comment:
All of these can be given a description. It isn't being used *yet* but will be soon as we expose this as part of the documentation system.

Reply:
This is now addressed. 👍🏼